src/generator/AutoRest.Java.Azure.Fluent/Model/MethodJvaf.cs

```diff
@@ -27,34 +28,49 @@ public override void Disambiguate()
 
             var methodType = GetMethodType(this);
             var originalName = Name;
-            string newName = null;
 
-            if (methodType == MethodType.ListBySubscription)
+            if (methodType != MethodType.Other)
             {
-                var otherListBySubMethods = this.MethodGroup.Methods.Where(x => GetMethodType(x as MethodJvaf) == MethodType.ListBySubscription);
-                if (otherListBySubMethods.Count() == 1)
+                string newName = null;
+
+                if (this.MethodGroup.Methods.Count(x => GetMethodType(x as MethodJvaf) == methodType) == 1)
                 {
-                    newName = "List";
+                    switch (methodType)
+                    {
+                        case MethodType.ListBySubscription:
+                            newName = "List";
+                            break;
+
+                        case MethodType.ListByResourceGroup:
+                            newName = "ListByResourceGroup";
+                            break;
+
+                        case MethodType.Delete:
+                            newName = "Delete";
+                            break;
+
+                        case MethodType.Get:
```

Comment:
Results in to 
https://github.com/Azure/azure-sdk-for-java/pull/1507/files#diff-a509efa70d61d3711b3bae5f0e28a4f8R222